src/App.tsx

```diff
@@ -1,8 +1,61 @@
+import { useState } from "react";
+import { getArrivalsGivenPostCode, type ArrivalInfo } from "../backend/fetchArrivals";
+
 function App() {
+  const [arrivalsData, setArrivalsData] = useState<ArrivalInfo[][] | string>();
+  const [text, setText] = useState("");
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setText(event.target.value);
+  };
+
+  async function handleSubmit() {
+    if(text === "") {
+      setArrivalsData("No postcode entered! Please enter a postcode first.");
+      return;
+    }
+    const response = await getArrivalsGivenPostCode(text);
+    console.log(response);
+    if(response === null) {
+      setArrivalsData("Arrivals not found");
+    }
+    else {
+      setArrivalsData(response);
+    }
+  }
+
   return (
-        <h1 className="text-3xl font-bold underline text-center text-cyan-600 m-4"
+        <>
+        <h1 className="text-3xl font-bold text-center text-cyan-600 m-8"
         >BusBoard</h1>
+        <div className="text-center border-black">
+          <input
+            className="p-2 m-12 text-center border-black"
+            type="text"
+            value={text}
+            placeholder="Type here..."
+            onChange={handleChange}
+          />
+          <button className="underline text-white px-4 py-2 bg-cyan-600 rounded" onClick={handleSubmit}>Search</button>
+        </div>
+        <div className="m-12 text-center">
+          {arrivalsData === undefined && (
+            <p>Please enter a postcode and click Search.</p>
+          ) }
+          {typeof arrivalsData === 'string' && (
+            <p className="text-red-500">{arrivalsData}</p>
+          )}
+          {Array.isArray(arrivalsData) && arrivalsData.length > 0 && arrivalsData.map((row, rowIndex) => (
```

Comment:
You could make this a variable to make it more readable
```suggestion
          {arrivalsDataIsNonEmpty && arrivalsData.map((row, rowIndex) => (
```